### DISCUSSION TITLE
Categorize variables display in plotting tab

### DISCUSSION THREAD

**Discussion by VicenteOG** [OpenModelica/OpenModelica#10423] [Q&A]
Hi!

I am developing a model that has a lot of variables and parameters so when I simulate it becomes a great mess in the variables window. I am hiding and protecting some of them so that they don't show, but still, the ones that are usefull for me are still plenty, so I would like to kind of categorize the variables in kind of "submodels" inside my model, so that if I want to explore a set of them that are related (ie, in the same "subcategory" that I define) I just expand the corresponding "submodel".

I know it might probably make more sense to make the different models, but I have a lot of interactions between the results and at the moment I am not 100% sure how to implement that, so I was thinking that having this possibility of categorize in "submodels" could be easy, so my question is: Is it possible to group some variables or parameters inside a model so that when simulating they are contained in a "submodel"?

Thanks in advance!

**Comment by casella** [comment]
I see three ways you can achieve this goal.

You can use [regular expressions in the variable filter](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omedit.html#output) to select subsets of interesting variables. If you have different sets, you can keep the corresponding regexp's in a test file, then copy and paste them in the variable filter input field when you need them. @adeas31 maybe we could enhance that input field so it remembers the last N expressions you used, so you can just click on them instead of doing copy-and-paste.

You can hard-wire this thing in the top-level model, e.g.
```modelica
Real V_001 = aaa.bb.c;
Real V_002 = aaa.dd.e.f;
...
Real W_001 = abc.df.d;
Real W_002 = def.ab.d;
....
```
In this case, just typing `V_` or `W_` in the variable filter brings out the variables of your interest.

Last, but not least, you can use the [DyMat](https://pypi.org/project/DyMat/) python package to read the .mat output files, and then build your own post-processing system in Python.
